src/main/java/com/e451/controller/TransactionController.java

```diff
@@ -0,0 +1,37 @@
+package com.e451.controller;
+
+import com.e451.domain.Transaction;
+import com.e451.service.TransactionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
```

Comment:
The boilerplate comment that IntelliJ adds should be deleted before you commit your code.